pkg/gatewayserver/io/basicstationlns/basicstationlns.go

```diff
@@ -135,9 +135,8 @@ func (s *srv) handleDiscover(c echo.Context) error {
 	ids := ttnpb.GatewayIdentifiers{
 		EUI: &req.EUI.EUI64,
 	}
-	ctx, ids, err = s.server.FillGatewayContext(ctx, ids)
+	_, ids, err = s.server.FillGatewayContext(ctx, ids)
```

Comment:
We should use the context, that's the point of filling it. We just shouldn't use it when this errors

Reply:
Yeah but the context is not actually needed here. `/router-info` only needs to check if the gateway is registered (or required to be registered). The actual returned context is not needed for this function.

Reply:
You don't know that at this point. We should be able to decorate the gateway context by GS, and we should not have specific frontends make assumptions about the need for that

Reply:
I don't get what you mean. Where in this bit of code do you see the need for using this returned context: https://github.com/TheThingsNetwork/lorawan-stack/blob/d2bd53d6d8b84700084d66b0375b591645542a85/pkg/gatewayserver/io/basicstationlns/basicstationlns.go#L139-L170?

Reply:
Taking the discussion offline